src/coreclr/jit/lower.h

```diff
@@ -217,9 +217,18 @@ class Lowering final : public Phase
         GenTree* oldUseNode = use.Def();
         if ((oldUseNode->gtOper != GT_LCL_VAR) || (tempNum != BAD_VAR_NUM))
         {
-            use.ReplaceWithLclVar(comp, tempNum);
+            GenTree* assign;
+            use.ReplaceWithLclVar(comp, tempNum, &assign);
+
             GenTree* newUseNode = use.Def();
             ContainCheckRange(oldUseNode->gtNext, newUseNode);
+
+            // We need to lower the LclVar and assignment since there may be certain
+            // types or scenarios, such as TYP_SIMD12, that need special handling
```

Comment:
We have existing handling for those node types which is otherwise missed, this leads to worse codegen or asserts: https://github.com/dotnet/runtime/blob/b443b8ccce174ef0e87183e21561c4ff13373cfd/src/coreclr/jit/lower.cpp#L283-L313

In the case of `GT_STORE_LCL_VAR` its expected that all `TYP_SIMD12` locals are widened to `TYP_SIMD16`, there is then other handling that exists for multiple register and constant handling that otherwise never happens for these introduced locals.

Reply:
Will do

Reply:
@echesakovMSFT no diffs for the `frameworks`, `benchmarks`, or `tests` pmi diffs from `jit-diff`

The "baseline" for the pre-SSE4.1 codegen fails due to the asserts that this PR fixes, ignoring those asserts the changes for the are just that the immediate and order of the operands on the shuffle are now correct.

Reply:
Thank you @tannergooding for confirming this.